randomActivityGen.py

```diff
@@ -91,6 +92,88 @@ def setup_city_gates(net: sumolib.net.Net, stats: ET.ElementTree, gate_count: in
         })
 
 
+def find_school_edges(net: sumolib.net.Net, num_schools):
+    edges = net.getEdges()
+
+    # Sort all edges based on their avg coord
+    edges.sort(key=lambda x: np.mean(get_edge_pair_centroid(x.getShape())))
+
+    # Split edges into n districts, with n being number of schools
+    district_size = int(np.ceil(len(edges) / num_schools))
+    districts = [edges[x:x + district_size] for x in range(0, len(edges), district_size)]
```

Comment:
So you are actually sorting by avg(x+y) and then splitting the city in slices to get discricts. I know, that you are aware; just putting it here for others.